### ISSUE TITLE
documentation updates for DR9

### ISSUE THREAD

**Issue by moustakas** [legacysurvey/legacysurvey#109]
I know we've been keeping track of changes on the wiki but I thought to create an issue as in previous DRs--
https://desi.lbl.gov/trac/wiki/DecamLegacy/DR9#ChangesfromDR8

A couple additional updates for DR9 are--
* https://github.com/legacysurvey/legacypipe/pull/469 (minor update to star cluster catalog);
* https://github.com/legacysurvey/legacypipe/pull/471 (less minor update to how we see Gaia stars).

**Comment by geordie666** [comment]
My preference is definitely GitHub rather than the wiki. So, I'd suggest putting issues here, if you like. I'll also link to them from the wiki at some point (either by copying them across or just linking to this issue).

**Comment by moustakas** [comment]
Sweeps have several data model changes, among them no more `FRACDEV` and new `SERSIC` column.

**Comment by moustakas** [comment]
* Globular cluster and planetary nebulae radii for 14 objects have been updated based on visual inspection--
  https://github.com/legacysurvey/legacypipe/pull/504

**Comment by moustakas** [comment]
The properties of the Local Group dwarfs in the LSLGA used for DR9 were taken from--
  https://vizier.u-strasbg.fr/viz-bin/VizieR-4

**Comment by moustakas** [comment]
A few more notes on the LG dwarfs (dumped from an email sent to chatter)--

```
Arjun and I went through the full catalog of LG dwarfs together and finalized
their category assignments:

1. Fornax & Sculptor -- move to globular cluster catalog (which will need to 
handle ellipse parameters now).
2. Exclude from LSLGA/any special handling (26 galaxies) -- DESI targeting may 
choose to mask these systems downstream but we shouldn't do anything about them.
3. Keep in LSLGA (including three new ones which weren't in the DR8 catalog) and 
pre-burn them (or maybe just pre-burn the largest ones).
```

**Comment by geordie666** [comment]
Addressed until this point in #116. I'll leave this open for additional documentation updates, though.

**Comment by moustakas** [comment]
Do we have documentation on where the Gaia catalogs came from (cf https://github.com/legacysurvey/legacypipe/issues/326)?

**Comment by geordie666** [comment]
Yes:

http://www.legacysurvey.org/dr8/external/#gaia-dr2

although you just reminded me that  I'll need to update the directories for the cfs directory structure (throughout)!

**Comment by dstndstn** [comment]
@geordie666 , I'm about to add some code where I had to make a choice:

We always keep Gaia sources, even if they would get cut by the model-selection criteria (notably, when they are so bright that they saturate our whole PSF postage stamp).  But since they have no model fits, we have to plug in *something* for their FLUX_[GRZ] values.  I made the decision to plug in our initial estimated-from-Gaia fluxes.  The FLUX_IVAR_[GRZ] columns will still be zero.

**Comment by geordie666** [comment]
Thanks @dstndstn: For clarity, are the estimated-from-Gaia fluxes simply the Gaia fluxes (in the Gaia filter system) shifted to our grz filter system? If so, is this using a flat SED? Or, do we perform some more sophisticated fitting than flat-SED even if we don't fit a anything resembling a proper model?

**Comment by dstndstn** [comment]
Hi Adam,

The estimated fluxes use Rongpu's polynomial estimates of Gaia-to-DECam
(DECam is used for both north and south surveys).

https://github.com/legacysurvey/legacypipe/blob/master/py/legacypipe/reference.py#L153

**Comment by dstndstn** [comment]
Hi Adam,

Changes to aperture photometry:

- apflux_[grz] are of the coadd images, masked by invvar=0.
- apflux_ivar_[grz] are, similarly, uncertainties on those masked coadds.
- apflux_masked_[grz] is a new column that measures what fraction of the pixels are masked; 1 means fully masked (ie, fully ignored; contributing 0 to the apflux).

The mask, being defined by invvar=0, ignores pixels that are masked in every overlapping exposure in that band.  Thus, the saturated cores and bleed trails of bright stars will be masked.  And if we have only one overlapping image, then cosmic rays and other mask bits will cause these pixels to be ignored.

**Comment by geordie666** [comment]
@dstndstn: There are other columns that could, in theory, be affected by these changes to aperture photometry. Can you confirm that the following are _not_ affected (are _not_ masked by `invvar=0`):

- `apflux_resid_[grz]` in the Tractor catalogs.
- `apflux_blobresid_[grz]` in the Tractor catalogs.
- `apflux` in the forced photometry files.

**Comment by dstndstn** [comment]
Hi Adam,
Thanks for thinking about this!  The apflux_resid and apflux_blobresid
columns already were being masked by invvar=0.  This doesn't affect the
forced photometry files.
cheers,
--dustin

**Comment by geordie666** [comment]
Remove `BRIGHTBLOB` from the DR9 data model.

**Comment by moustakas** [comment]
(Another) update on how we force point sources from Gaia--
https://github.com/legacysurvey/legacypipe/issues/570#issuecomment-637671577

**Comment by geordie666** [comment]
This is a note to myself that I've addressed comments in this issue thus far in #116 and on [the test page](http://faraday.uwyo.edu/~admyers/scratch/legacysurvey/output/).

**Comment by geordie666** [comment]
The description of `FIBERFLUX` is "Predicted x-band flux within a fiber from this object in 1 arcsec Gaussian seeing", and it does not mention the fiber diameter in the calculation. Need to add that the fiber diameter is 1.5":

https://github.com/legacysurvey/legacypipe/blob/master/py/legacypipe/runbrick.py#L1839

(see also #118).

**Comment by geordie666** [comment]
Make sure to document the (new) WISE columns in the tractor catalogs. @djschlegel notes that the sizes and rational for the aperture photometry is [here](https://github.com/legacysurvey/legacypipe/issues/447) and that the `WISE_X` and `WISE_Y` positions refer to the WISE coadds that matches `WISE_COADD_ID`, rather than to the images in the legacypipe coadd directory.

@dstndstn: can you comment on how to locate sources on the images in the `legacysurvey-*-image-W1.fits.fz` files? Are the tan projections in the header WCS identical to `BX*342/3600` and `BY*342/3600`?

Columns to make sure to add are:

```
   APFLUX_W1       FLOAT     Array[5]
   APFLUX_W2       FLOAT     Array[5]
   APFLUX_W3       FLOAT     Array[5]
   APFLUX_RESID_W1 FLOAT     Array[5]
   APFLUX_RESID_W2 FLOAT     Array[5]
   APFLUX_RESID_W3 FLOAT     Array[5]
   APFLUX_RESID_W4 FLOAT     Array[5]
   APFLUX_IVAR_W1  FLOAT     Array[5]
   APFLUX_IVAR_W2  FLOAT     Array[5]
   APFLUX_IVAR_W3  FLOAT     Array[5]
   APFLUX_IVAR_W4  FLOAT     Array[5]
   WISE_X          FLOAT           17.8082                                         
   WISE_Y          FLOAT           5.42195                                         
```

**Comment by geordie666** [comment]
@djschlegel also reports (apropos the `APFLUX_*` columns):

```
The optical and WISE images are both tangential projections
with central pixel scale of 0.262 and 2.75 arcsec/pix.  With the WCS
choices, the transformation from BX,BY to the WISE image pixels would be
   XWISE = BX * 0.262/2.75 - 0.94327
   YWISE = BY * 0.262/2.75 - 0.94327
```

**Comment by geordie666** [comment]
Remember to update the full suite of `RELEASE` numbers that have been used in DR9 testing.

**Comment by geordie666** [comment]
Spin the cutout service to its own page under the "viewer" tab, so that it's easier to find. Make sure to provide a link to [the list of allowed urls](http://legacysurvey.org/viewer/urls).

**Comment by geordie666** [comment]
See https://github.com/legacysurvey/legacypipe/issues/593 for documentation of a further update to the data model of the tractor files.

**Comment by geordie666** [comment]
Document file of star masks in `/dr9/external/`

**Comment by geordie666** [comment]
Note which columns are missing for the SGA files (and note that we drop those columns deliberately in the pre-burn...and that the information is still retained in the tractor catalogs).

**Comment by moustakas** [comment]
Pre-burned large-galaxy coadds do not have the sky template subtraction (see https://github.com/legacysurvey/legacypipe/issues/609).

**Comment by geordie666** [comment]
@djschlegel notes that the OOW images are in the same units as `1/SKYRMS**2` in the survey-ccds files _but_ that these images need to be multiplied by both `gain` and `exptime` to get electrons (for 90prime+mosaic).  We should have some minimal documentation on this at the end of the “Files” page.

**Comment by geordie666** [comment]
Related to the comment on June 29th, @dstndstn notes:

I think David already worked out the transformation for pixel coordinates in the `legacysurvey-BRICK-image-W1.fits.fz` files compared to (zero-indexed) `BX`, `BY` coordinates, which apply to the optical coadds:
```
XWISE = BX * 0.262/2.75 - 0.94327
```
This peculiar offset comes from the fact that both WCS headers share the same (brick) center, which is pinned to the center of each image (where the WISE image size is chosen by rounding DOWN (3600 * 0.262 / 2.75).  So it's possible to write David's expression as (including FITS +-1 pixels):
```
XWISE = (BX - 1 - 1800.5) * 0.262/2.75 + 171.5 - 1
```

**Comment by geordie666** [comment]
Just a note that everything in this issue (so far) was addressed in #116. I'll leave this open in case of further updates, though, until DR9 is final.

**Comment by geordie666** [comment]
I'm closing this now, given the near-final website in #125. Everything in this thread has been addressed. We should open a new issue if we need further changes after the near-final version of DR9 is deployed.
